### PR TITLE
go scripts dependency update

### PR DESCRIPTION
**Pull request by kroos47**

### CODE REVIEW

**Review by MiguelDD1 — APPROVED** [review]

File: tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go [hunk 1]
```diff
@@ -45,18 +46,20 @@ func main() {
 
 	// Set data and appID according to need
 	size := 0
-	if config.Size == 0  || config.Size > -1 {
+	if config.Size == 0 || config.Size > -1 {
 		size = 100
 	}
-	sub_data , _:=  extrinsics.RandToken(size)
+	sub_data, _ := extrinsics.RandToken(size)
+	t_data := sub_data
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go:53]: Why is this needed?

File: tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go [hunk 2]
```diff
@@ -133,3 +138,30 @@ func main() {
...
+		if ext.Method.CallIndex.SectionIndex == 29 && ext.Method.CallIndex.MethodIndex == 1 {
+			arg := ext.Method.Args
+			str := string(arg)
+			slice := str[2:]
+			fmt.Println("string value", slice)
+			fmt.Println("data", data)
+			if slice == data {
+				fmt.Println("Data found in block")
+			}
+			// buf := bytes.NewBuffer(ext.Method.Args)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go:157]: Can we remove these comments?

File: tests/gsrpc/extrinsics/test_bal/availaccountbalance.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
+package main
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/test_bal/availaccountbalance.go:1]: There are multiple balance transfer scripts. Can we remove/consolidate some of them? This one seems to be redundant, especially bcoz the only unique part of creating appID is probably commented out. Let us remove this script if it is not showing anything new and instead put this as part of one of the tests inside gsrpc, if possible.

File: tests/gsrpc/extrinsics/encode/encode.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
+// You can edit this code!
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/encode/encode.go:1]: Can we remove the comments?

File: tests/gsrpc/extrinsics/encode/encode.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+								AppId: AppId(NewUCompactFromUInt(2)),
+								Start: NewUCompactFromUInt(10),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	x, err := EncodeToHex(exampleHeader)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/encode/encode.go:68]: Not sure why this example is important in the repo. This probably should be part of test inside gsrpc.

**Review by prabal-banerjee — CHANGES_REQUESTED** [review]

File: tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go [hunk 6]
```diff
@@ -45,10 +46,13 @@ func main() {
 
 	// Set data and appID according to need
 	size := 0
-	if config.Size == 0  || config.Size > -1 {
+	if config.Size <= 0 {
 		size = 100
+	} else {
+		size = config.Size
 	}
```
Comment (aterentic-ethernal) [inline tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go:53]: Maybe:
```
size := 100
if config.Size > 0 {
    size = config.Size
}
```

File: tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go [hunk 7]
```diff
@@ -114,7 +115,7 @@ func main() {
 		panic(err)
 	}
 	size := 0
-	if config.Size == 0  || config.Size > -1 {
+	if config.Size <= 0 {
 		size = 100
 	}
```
Comment (aterentic-ethernal) [inline tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go:120]: This is probably a bug, since `config.Size` is ignored, we should probably repeat same code from `dataSubmitAndWatch.go`, or, if possible, assign default value to `config.Size` and just use it.

File: tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go (hunk 6 above)
Comment (kroos47) [reply tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go:53]: well suppose the config.size is missing just wanted to take a default size of 100.

**Review by prabal-banerjee — APPROVED** [review]
